### DISCUSSION TITLE
Need help installing IPOPT with Intel MKL

### DISCUSSION THREAD

**Discussion by maddielynklein** [coin-or/Ipopt#441] [Q&A] [answered]
Hello, I'm trying to configure and install IPOPT to use Intel MKL on mac OS X. I have Intel MKL installed in the usual /opt/intel/oneapi/mkl and $MKLROOT is set to /opt/intel/oneapi/mkl/latest. When I run the IPOPT configure script it is successful and it looks like it found the MKL libraries based on these lines of output
`configure: pkg-config path is "/usr/local/lib/pkgconfig:/opt/intel/oneapi/mkl/latest/bin/pkgconfig"`
`checking for function dsyev_ in -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lm ... yes`

However when I run `sudo make install` it fails at this point claiming it can not find the MKL library
`Making install in src/Interfaces`
`/Library/Developer/CommandLineTools/usr/bin/make  install-am`
`  CXXLD    libipopt.la`
`ld: library not found for -lmkl_intel_lp64`
`clang: error: linker command failed with exit code 1 (use -v to see invocation)`
`make[2]: *** [libipopt.la] Error 1`
`make[1]: *** [install] Error 2`
`make: *** [install-recursive] Error 1`

Any help is greatly appreciated, thanks!

**Comment by svigerske** [comment, marked as answer]
Probably the library search path in the sudo environment was different from the one of configure, and then it didn't find the MKL lib anymore.
Normally there should be no need to link the Ipopt lib in the `make install` step, since that should have happened in the `make` step, but something I see libtool saying that it relinks the lib. Maybe try `sudo -E`.

**Reply by maddielynklein** [reply]
That made it work, thanks so much!!
